sycl/include/CL/sycl/accessor.hpp

```diff
@@ -730,83 +733,245 @@ class accessor<DataT, Dimensions, AccessMode, access::target::local,
 // Image accessor
 template <typename DataT, int Dimensions, access::mode AccessMode,
           access::target AccessTarget, access::placeholder IsPlaceholder>
-class image_accessor {
-  static_assert(AccessTarget == access::target::image ||
-                    AccessTarget == access::target::host_image ||
-                    AccessTarget == access::target::image_array,
+class image_accessor
+#ifndef __SYCL_DEVICE_ONLY__
+    : public detail::AccessorBaseHost {
+  size_t MImageSize;
+  size_t MImageCount;
+#else
+{
+  /*
+  // TODO: Define the datatype here based on Dimensions, AccessMode and
+  AccessTarget.
+  __ocl_image[Dim]d[array/non_array]_[AM]_t MImage;
+  __init(__ocl_imagexx_t Image) { MImage = Image; }
+  */
+#endif
+  constexpr static bool IsHostImageAcc =
+      (AccessTarget == access::target::host_image);
+
+  constexpr static bool IsImageAcc = (AccessTarget == access::target::image);
+
+  constexpr static bool IsImageArrayAcc =
+      (AccessTarget == access::target::image_array);
+
+  constexpr static bool IsImageAccessAnyWrite =
+      (AccessMode == access::mode::write ||
+       AccessMode == access::mode::discard_write);
+
+  constexpr static bool IsImageAccessRead = (AccessMode == access::mode::read);
+
+  static_assert(IsImageAcc || IsHostImageAcc || IsImageArrayAcc,
                 "Expected image type");
-  // TODO: Check if placeholder is applicable here.
+
+  static_assert(IsPlaceholder == access::placeholder::false_t,
+                "Expected false as Placeholder value for image accessor.");
+
+  static_assert(
+      AccessMode == access::mode::read || AccessMode == access::mode::write ||
+          AccessMode == access::mode::discard_write,
+      "Access modes can be read/write/discard_write for image accessor.");
+
+  static_assert(Dimensions > 0 && Dimensions <= 3,
+                "Dimensions can be 1/2/3 for image accessor.");
+
+  template <bool B, class T = void>
+  using enable_if_t = typename std::enable_if<B, T>::type;
+
 public:
   using value_type = DataT;
   using reference = DataT &;
   using const_reference = const DataT &;
 
-  /* Available only when: accessTarget == access::target::host_image */
+  // image_accessor Constructors.
+
+  // Available only when: accessTarget == access::target::host_image
   // template <typename AllocatorT>
   // accessor(image<dimensions, AllocatorT> &imageRef);
-  /* Available only when: accessTarget == access::target::image */
+  template <typename AllocatorT, int Dims = Dimensions,
+            typename = enable_if_t<(Dims > 0 && Dims <= 3) && IsHostImageAcc>>
+  image_accessor(image<Dims, AllocatorT> &ImageRef, int ImageElementSize)
+#ifdef __SYCL_DEVICE_ONLY__
+  { // No constructor needed for the device side.
```

Comment:
This is actually a constructor you are defining here...
Do you mean `= delete;` instead?

Reply:
Or just change the comment to fit the goal.

Reply:
I had similar issue with buffer accessor few months ago, removing '{ }' did not work because the constructor was needed even for DEVICE because the filtering of DEVICE code happened much later.
I don't remember the details.
Garima, did you try " = delete", or having ';' instead of '{ }' ?

Reply:
It reports the error when i try to use instantiate it.
error: call to deleted constructor of 'image_accessor